### PR TITLE
51 automate renv creation

### LINKED ISSUE #34: Remove Ross from propagation
@rossfarrugia would like to be taken off these propagation PRs

https://github.com/pharmaverse/admiralci/blob/4f0d6f74e9ae96a40fbb6f90b192dbefe6c87d9f/.github/workflows/renv.yml#L53

### LINKED ISSUE #39: Documentation:  automate a check to throw an error if library() statement found in testthat

### LINKED ISSUE #51: Automate `renv.lock` creation
The creation of `renv.lock` should be automated. A script should be implemented which creates `renv.lock` fulfilling the following requirements:
- The lock file should contain all packages which are imported or suggested by any of the admiral packages. The version of the packages should be the one which was available when the R version we are using for development was released. At the moment we use R 4.0. I.e., we would store the version in the lock file which was available when R 4.0 was released.
- There should be an option to use newer versions for some packages. For example, we use a newer version of pkgdown. The packages and versions should be stored in a separate file. It is read in by the script. Then the exceptions are clear and easy to maintain.
- The admiral packages should not be in the lock file because these are handled by staged_dependencies.

### LINKED ISSUE #63: The Microsoft R Application Network website will be shut down on July 1st, 2023.
[The Microsoft R Application Network website will be shut down on July 1st, 2023](https://techcommunity.microsoft.com/t5/azure-sql-blog/microsoft-r-application-network-retirement/ba-p/3707161), and we should probably migrate to [Posit Package Manager](https://packagemanager.posit.co/).

### PR DESCRIPTION
**Pull request by galachad**
Close #51
Close #34
Close #39 
Close #63 
Close pharmaverse/admiral#1799 (together with propagation)

- [x] Gen renv.lock automaticity.
- [x] Repository commit and push
- [x] Better rules for triggering this job for PR (to review with cicdguy)
- [x] Verify the logic behind renv.lock
- [x] Update docs about generating renv in admiraldev
- [x] Verify renv.lock propagation

### CODE REVIEW

**Comment by galachad** [comment]
Hi @cicdguy I would like to use `workflow_run` but I can not test it. The `workflow_run` will be trigger only from default branch. Do you have other proposition?

Example:
```
 workflow_run:
    workflows: ["Generate renv.lock"]
    types:
      - completed
```

**Comment by bundfussr** [comment]
Even if we use container-based CIs, I would keep the renv.lock for some time because not all contributors may be able to use containers.

**Comment by bms63** [comment]
I think we should have both as I am not convinced that we can use containers on our GSK laptops.  

@kaz462 can Bayer use containers on company laptops?

**Comment by kaz462** [comment]
> @kaz462 can Bayer use containers on company laptops?

I might need to contact IT for installing Docker on Bayer's laptop, but right now I'm using my own PC for admiral development, which can use containers.

**Comment by galachad** [comment]
Hi @cicdguy, look like all dependence PR was merged, so please take a look, and we can merge this one too.

**Comment by galachad** [comment]
Hi @bms63, could you take a look? So we will be able to close this stage of propagation.

File: .github/workflows/renv.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ on:
     paths:
       - renv.lock
       - renv/**
+  workflow_run:
+    workflows: ["Generate renv.lock"]
+    types:
+      - completed
```
Comment (bms63) [inline .github/workflows/renv.yml:17]: @galachad can we add admiralvaccines to this list please

**Review by bms63 — APPROVED** [review]

**Pull request merged**